### PR TITLE
Sim test

### PR DESCRIPTION
**Pull request by FreyJo**
Hey guys,

I modified the `sim_wt_model_nx3.c` example, such that all currently implemented integrators can be used.
Using this example, I wrote the `sim_test.cpp` function. In this test a reference solution is created. Subsequently all integrators are called with different `num_steps`.
The outputs (simulation, forward and adjoint sensitivities) are compared against the reference solution.
For `LIFTED_IRK` and `NEW_LIFTED_IRK`, adjoint sensitivity propagation does not work (yet), thus this is excluded from the test and a todo has been added.

### CODE REVIEW

**Comment by dkouzoup** [comment]
Hey @FreyJo, thanks for start working on the sim tests! You need to (a) pull the latest master to resolve conflicts and (b) fix travis, before we merge

File: examples/c/sim_wt_model_nx3.c [hunk 1]
```diff
@@ -286,6 +286,8 @@ int main()
 			case 0:
 				// ERK
 				opts->ns = 4; // number of stages in rk integrator
+				opts->sens_adj = true;
```
Comment (dkouzoup) [inline examples/c/sim_wt_model_nx3.c]: you mean _re_activated? ;)

**Comment by FreyJo** [comment]
Finally finished the `sim_test`.
I wrote the issue #299 which refers to the not covered cases in the test.
Should be ready to merge now!

**Comment by giaf** [comment]
You should also write the unit test to detect NaN, as described in https://github.com/acados/acados/issues/300

**Comment by FreyJo** [comment]
@giaf thanks for your hints before!
Now all the valgrind warnings are gone :+1:

**Comment by giaf** [comment]
Now on my PC I get this error when running the unit tests:

In file included from /home/gianluca/acados/external/catch/include/catch.hpp:34:0,
                 from /home/gianluca/acados/test/sim/sim_test.cpp:32:
/home/gianluca/acados/test/sim/sim_test.cpp: In function ‘void ____C_A_T_C_H____T_E_S_T____0()’:
/home/gianluca/acados/test/sim/sim_test.cpp:539:50: error: ‘isnan’ was not declared in this scope
                         REQUIRE(isnan(out->xn[jj]) == 0);
                                                  ^

Maybe the C++ compiler does not like isnan (at least by default, maybe need to use some flags? @roversch ).
@FreyJo  @dkouzoup @zanellia is this working on your PCs?

Commenting out the checks with isnan everything works fine (the segfault is gone too).

**Comment by dkouzoup** [comment]
@FreyJo do you include the math.h header?

**Comment by FreyJo** [comment]
@dkouzoup yes, it is included in the `sim_test.cpp`

**Comment by dkouzoup** [comment]
I confirm that it works fine on my PC, let's try to fix the issue on @giaf's PC tomorrow and then merge finally ;)

**Comment by dkouzoup** [comment]
shall we merge @giaf?

**Pull request merged**